modules/apps/data-engine/data-engine-rest-test/src/testIntegration/java/com/liferay/data/engine/rest/resource/v2_0/test/DataDefinitionResourceTest.java

```diff
@@ -620,6 +625,55 @@ public void testPostSiteDataDefinitionByContentType() throws Exception {
 	@Override
 	@Test
 	public void testPutDataDefinition() throws Exception {
+		String originalName = PrincipalThreadLocal.getName();
+
+		PrincipalThreadLocal.setName(TestPropsValues.getUserId());
+
+		Queue<Long> queue = new LinkedList<>();
+
+		ReflectionTestUtil.setFieldValue(
+			_dataDefinitionResource, "_ddmStructureLocalService",
+			ProxyUtil.newProxyInstance(
+				DDMStructureLocalService.class.getClassLoader(),
+				new Class<?>[] {DDMStructureLocalService.class},
+				(proxy, method, arguments) -> {
+					if (Objects.equals(method.getName(), "updateStructure")) {
+						queue.add((Long)arguments[1]);
+					}
+
+					return method.invoke(_ddmStructureLocalService, arguments);
+				}));
+
+		DataDefinition dataDefinition =
+			dataDefinitionResource.postSiteDataDefinitionByContentType(
+				testGroup.getGroupId(), _CONTENT_TYPE,
+				DataDefinition.toDTO(
+					DataDefinitionTestUtil.read(
+						"data-definition-fieldset-1.json")));
+
+		dataDefinitionResource.postSiteDataDefinitionByContentType(
+			testGroup.getGroupId(), _CONTENT_TYPE,
+			DataDefinition.toDTO(
+				DataDefinitionTestUtil.read(
+					"data-definition-fieldset-2.json")));
+
+		dataDefinitionResource.postSiteDataDefinitionByContentType(
+			testGroup.getGroupId(), _CONTENT_TYPE,
+			DataDefinition.toDTO(
+				DataDefinitionTestUtil.read(
+					"data-definition-fieldset-3.json")));
+
+		_dataDefinitionResource.setContextUser(TestPropsValues.getUser());
+
+		_dataDefinitionResource.putDataDefinition(
+			dataDefinition.getId(),
+			com.liferay.data.engine.rest.dto.v2_0.DataDefinition.toDTO(
+				dataDefinition.toString()));
+
+		Assert.assertEquals(3, queue.size());
```

Comment:
Can you put this code in the Try/Finally?
In the Finally block > PrincipalThreadLocal.setName(originalName);
This would guarantee to execute the Finally block regardless of whether an exception occurs or not. 
This ensures that PrincipalThreadLocal.setName(originalName) is always called to set it back to its original value, preventing potential side effects in subsequent tests.